libraries/standard/http/src/http_client.c

```diff
@@ -112,6 +100,21 @@ static HTTPStatus_t _addHeader( HTTPRequestHeaders_t * pRequestHeaders,
         pRequestHeaders->headersLen += toAddLen;
         returnStatus = HTTP_SUCCESS;
     }
+    else
+    {
+        /* Restore the length of the trailing line separator. */
+        if( hasTrailingLine == 1u )
```

Comment:
This update operation wouldn't be required if the `headersLen` update is only dealt within in the happy-path branch above (as I suggested in [this comment](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/892/files/b12faa800ad2f117faf5f2318797173a02a4d31d..4acdde70d7b7409251a1c9fd7d2645915c7b9735#r414869709))